app/src/main/java/com/example/tasky/core/util/StateObserver.kt

```diff
@@ -22,4 +22,13 @@ fun ObserveBoolean(observer: Boolean?, onTriggered: () -> Unit) {
             onTriggered()
         }
     }
+}
+
+@Composable
+fun ObserveString(observer: String?, onTriggered: (String) -> Unit) {
+    LaunchedEffect(key1 = observer) {
+        if (observer != null) {
+            onTriggered(observer)
+        }
+    }
```

Comment:
Why not make this generic, so you don't need ObserveString, ObserveBoolean, ObserveInt, etc.? In general, I probably wouldn't even make this a function just to not need an if-check, but if you like that, this could be a function that works for all types:
```
@Compoable
fun <T> Observe(observer: T?, onTriggered(T) -> Unit) {
    LaunchedEffect(key1 = observer) {
        if (observer != null) {
            onTriggered(observer)
        }
    }
}
```